.circleci/config.yml

```diff
@@ -298,18 +299,43 @@ jobs:
       - run:
           name: Validate Swagger definitions
           working_directory: ~/workbench/api
-          command: ./project.rb validate-swagger --project-prop verboseTestLogging=yes
+          command: |
+            if [ "$CIRCLE_NODE_INDEX" == 0 ]; then
+              ./project.rb validate-swagger --project-prop verboseTestLogging=yes
+            fi
       - run:
           name: Run Java unit tests
           working_directory: ~/workbench/api
-          command: ./project.rb test
+          # See: https://circleci.com/docs/2.0/language-java/#sample-configuration
+          # Short explanation on commands:
+          #   Find all unit tests recursively in "api/src/test/java/" directories.
+          #   Strips off "src/test/java/" substring;
+          #   Replaces all path separator "/" with ".";
+          #   Removes file extensions.
+          #   Append "--tests" in front of file name.
+          command: |
+            CLASSNAMES=$(circleci tests glob "src/test/java/**/*Test.java" "src/test/java/**/*Test.kt" \
+              | cut -c 1- \
+              | sed 's@src/test/java/@@' \
+              | sed 's@/@.@g' \
+              | sed 's/\.[^.]*$//' \
+              | circleci tests split --split-by=timings --timings-type=classname --index=$CIRCLE_NODE_INDEX)
+            GRADLE_ARGS=$(echo $CLASSNAMES | awk '{for (i=1; i<=NF; i++) print "--tests", $i}')
+            java -version && gradle -v && gradle test $GRADLE_ARGS
```

Comment:
nit: Can you move the java & gradle version checks to their own step if they're necessary? I guess I was thinking that's specified by the image, so maybe we don't need it here too?

Reply:
removed `java -version && gradle -v` substring. I thought it would be good to have for info, but you're right, it is already in docker image. No need to show here.